dev/io.openliberty.microprofile.metrics.5.0.internal/src/io/openliberty/microprofile/metrics50/SharedMetricRegistries.java

```diff
@@ -73,6 +88,40 @@ public void associateMetricIDToApplication(MetricID metricID, String appName, Me
         }
     }
 
+    public void setAppNameResolver() {
+        /*
+         * Unsuccessful activation. SmallRye and Micrometer classes were not loaded
+         * properly.
+         */
+        if (!smallryeMetricsCDIMetadata.isSuccesfulActivation()) {
+            return;
+        }
+
+        ClassLoader baoClassLoader = Util.BUNDLE_ADD_ON_CLASSLOADER;
+        Object applicationNameResolverProxy = Proxy.newProxyInstance(baoClassLoader,
```

Comment:
Create an implementation of the (SmallRye) `ApplicationNameResolver` interface using reflection.